src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,475 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Animated, View, TouchableOpacity, TouchableHighlight} from 'react-native';
+import Interactable from 'react-native-interactable';
+import {BaseComponent, Constants, Colors, Typography} from '../../../src';
+
+
+const SCALE_POINT = 72; // scaling content style by height
+const MIN_LEFT_MARGIN = 28;
+const DEFAULT_ICON_SIZE = 24;
+const MIN_ITEM_WIDTH = 43; // NOTE: this is the min for the input ranges calc!
+const ITEM_BG = Colors.blue30;
+const ITEM_PADDING = 12;
+const BLEED = 15;
+const ITEM_PROP_TYPES = {
+  width: PropTypes.number,
+  background: PropTypes.string,
+  text: PropTypes.string,
+  icon: PropTypes.number,
+  onPress: PropTypes.func,
+};
+
+/**
+ * @description: Interactable Drawer component
+ * @extendslink: 
+ */
+export default class Drawer extends BaseComponent {
+  static displayName = 'Drawer';
+
+  static propTypes = {
+    /**
+     * The drawer top layer's damping
+     */
+    damping: PropTypes.number,
+    /**
+     * The drawer top layer's tention
+     */
+    tension: PropTypes.number,
+    /**
+     * The bottom layer's items to appear when opened from the right
+     */
+    rightItems: PropTypes.arrayOf(PropTypes.shape(ITEM_PROP_TYPES)),
+    /**
+     * The bottom layer's item to appear when opened from the left (a single item)
+     */
+    leftItem: PropTypes.shape(ITEM_PROP_TYPES),
+    /**
+     * Whether to give the items equal width (the max width)
+     */
+    equalWidths: PropTypes.bool,
+    /**
+     * The color for the text and icon tint of the items
+     */
+    itemsTintColor: PropTypes.string,
+    /**
+     * The items' icon size
+     */
+    itemsIconSize: PropTypes.number,
+    /**
+     * The items' text style
+     */
+    itemsTextStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+    /**
+     * Press handler (will also close the drawer)
+     */
+    onPress: PropTypes.func,
+  };
+
+  static defaultProps = {
+    damping: 1 - 0.6,
+    tension: 300,
+    itemsTintColor: Colors.white,
+    itemsIconSize: DEFAULT_ICON_SIZE,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.deltaX = new Animated.Value(0);
+    this.leftItemJSON = JSON.stringify(this.props.leftItem);
+    this.rightItemsJSON = JSON.stringify(this.props.rightItems);
+
+    this.state = {
+      inMotion: false,
+      position: 1,
+      width: Constants.screenWidth,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.leftItemJSON !== JSON.stringify(nextProps.leftItem) || 
```

Comment:
this comparison will always be against the initial value of the buttons. 

What if the value of buttons was initially X then changed to Y and then again to X, the second time it will compare it, it will compare X to X instead of Y to X.